src/test/java/mate/academy/security/AuthenticationServiceImplTest.java

```diff
@@ -0,0 +1,72 @@
+package mate.academy.security;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.RoleService;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+public class AuthenticationServiceImplTest {
+    private AuthenticationService authenticationService;
+    private RoleService roleService;
+    private UserService userService;
+    private PasswordEncoder passwordEncoder;
+    private User user;
+
+    @BeforeEach
+    void setup() {
+        user = new User();
+        user.setEmail("test@test.com");
+        user.setPassword("test");
+        Role role = new Role();
+        role.setRoleName(Role.RoleName.USER);
+        roleService = Mockito.mock(RoleService.class);
+        userService = Mockito.mock(UserService.class);
+        passwordEncoder = Mockito.mock(PasswordEncoder.class);
+        Mockito.when(roleService.getRoleByName("USER")).thenReturn(role);
+        authenticationService =
+                new AuthenticationServiceImpl(userService, roleService, passwordEncoder);
```

Comment:
Differently to DAO layer where we want reset state our services here don't hold any state. So there is no reason to recreate objects for each test.
Please move to BeforeAll method or initialize objects during declaration
Please recheck all classes